### ISSUE TITLE
Where does this source originate?

### ISSUE THREAD

**Issue by fluffynukeit** [simonjwright/ASIS#2]
Hi and thanks for providing the ASIS-for-gnat source in this repo.  Where does the source code actually originate?  For this and also your gnat_util repo.  Searching online for "ASIS-for-GNAT" either leads me back to here or to dead ends.  I'd like to have better control over the versions of ASIS for gnat that are available to me, e.g. use a version for gcc 10.2 instead of 10.1 that is tagged in your repo.

My motivation is that I'm creating a nix flake to build the typical suite of Ada and SPARK tools from source. Right now, I'm using your ASIS 10.1 tag instead of the exactly matching 10.2 version I would require.  It's compiling and executes the tools but I'd feel better if I could use the exact matching version directly from the source.  https://github.com/fluffynukeit/adaspark

Thanks much!

**Comment by simonjwright** [comment]
One thing you may not be aware of is that AdaCore are dropping ASIS support (I guess if you were a paying customer it might be different).
This shows up because neither GNAT CE 2020 nor FSF GCC 11 supports the `-t` switch that requests generation of the `.adt` files that GNAT ASIS uses.

The sources in this repo come from the AdaCore site, and I’ve imported them under the `upstream` branch. The last one was `asis-gpl-2019-src`. The changes in the `fsf` branch are those needed to adapt the upstream sources to FSF GCC; for example, differences in the naming of compiler releases, which if not catered for would mean that ASIS would regard any tree as invalid.

The reason for sticking at 10.1 is that I only regenerate this for major releases, tied to my macOS compiler bundles;  there’s usually very little change in the Ada sources between point releases. If I’d needed to build 10.2, I’d’ve tried the 10.1 release here, and if it compiled OK, called it good.

The [gnat_util](https://github.com/simonjwright/gnat_util) repo is needed for ASIS, and used to be needed for - I think- one of the GNATCOLL components, maybe [gnatcoll-db](https://github.com/AdaCore/gnatcoll-db)’s `Xref`. Anyway, it only provides one source file of its own (`prefix.c`), the rest come from the compiler sources (see `compiler_files`).

**Comment by fluffynukeit** [comment]
Thanks for the explanation.  I am aware of ASIS support dropping.  I am ok with it as long as I have a reasonably modern compiler that can run gnattest, so gcc 10 makes sense for me.  I did indeed try the 10.1 branch of yours with success when running my nix flake against GCC/gnat 10.2, so everything built and presumably works ok with the minor version mismatch.

Can you link to the source locations?  That was my original question, apologies for being unclear.  I have not been able to find it.  All roads lead back to you and your github.  Maybe the location is well-known to Ada aficionados, but as a newb I have not found it.  Many historical links point to libre.adacore.com, but it now redirects to community edition home page.

**Comment by simonjwright** [comment]
Oh, sorry.

2017: https://community.download.adacore.com/v1/369101e6b0d9e777b9bcae821ddb68567138a95f?filename=asis-gpl-2017-src.tar.gz

2018:
https://community.download.adacore.com/v1/c338044768412af787c8cff13c0d952ed688df11?filename=asis-gpl-2018-20180524-src.tar.gz

2019:
https://community.download.adacore.com/v1/52c69e7295dc301ce670334f8150193ecbec580d?filename=asis-2019-20190517-18AB5-src.tar.gz

In case those don’t work for you,

* Go to https://www.adacore.com/community
* click on Download
* at the bottom right, click on "More packages, platforms, versions and sources >"
* in the "Select your platform" combo, select "86 GNU Linux" (because it has the most complete set of sources)
* select e.g. 2017 in the right-hand combo
* under "GNAT GPL Ada", select "Sources"
* ... there it is!

You’ll see I took a liberty with the names of my commits, not wholly helpful: but excuse is, the names I used were the names that the 2017 & 2018 tarballs unpack into :-) so I followed the same pattern for the 2019 version.
